fem/nonlininteg.hpp

```diff
@@ -79,13 +79,28 @@ class NonlinearFormIntegrator
                                       ElementTransformation &Tr,
                                       const Vector &elfun, Vector &elvect);
 
+   /// Perform the local action (mixed) of the NonlinearFormIntegrator
+   virtual void AssembleElementVector2(const FiniteElement &trial_fe,
```

Comment:
What would be the need for this function, and how would it be different from the `AssembleFaceVector` below?

Reply:
I just realized (multiple years ago 😄 ) there is no method for mixed action, i.e., different trial and test space. It is not necessarily related to DGs, but any elements, which you want to integrate over the volume (unlike `AssembleFaceVector` for faces).

Reply:
And it took you this long to fix it!! 😂
The existing `AssembleFaceVector(const FiniteElement &, const FiniteElement &,...)` can also be used to handle diff. trial & test
To avoid any confusion, lets  rename`AssembleFaceVector2` to something that explicitly says mixed (e.g., `AssembleFaceVectorMixed` and maybe a function like `AssembleFaceVector(const FiniteElement &, const FiniteElement &,..., bool mixed_form)` that calls the right assembly (PG/BG)

Reply:
I was leaving the commits in/on my fork to properly age them 😄 .
Do you mean `AssembleElementVector2`? I do not know, it follows the naming of `BilinearFormIntegrator`, which has `AssembleElementMatrix` (single space) and `AssembleElementMatrix2` (two spaces, i.e., mixed). It is true the current methods can be used for mixed integration as well, but you need to give the integrators access to the finite element space used for the trial (or test) functions, which is only a little bit less nasty than using `AssembleElementMatrix()` instead of `AssembleElementMatrix2()`, so it should not be encouraged anyhow to be done by users.